libsmtutil/SolverInterface.h

```diff
@@ -305,6 +307,76 @@ class Expression
 		);
 	}
 
+	static Expression mkAnd(std::vector<Expression> _args)
+	{
+		smtAssert(!_args.empty(), "");
+		auto sort = _args.front().sort;
+		smtAssert(
+			ranges::all_of(
+				_args,
+				[&](auto const& _expr){ return _expr.sort->kind == sort->kind; }
+			),
+			""
+		);
```

Comment:
Maybe it would be better to define a helper for this check?

```suggestion
		smtAssert(sameSort(_args));
```

Reply:
done